File: data structures/queue/queue.py
class queue :

    def __init__(self,n):
        self.list = [0]*n
        self.size = n
        self.front = -1
        self.rear = -1
    
    def insert(self,a):
        if(self.front==-1):
            self.front+=1
            self.rear+=1
            self.list[self.front]=a
        elif self.rear==self.size-1 :
            print("the queue is full")
        else :
             self.rear+=1
             self.list[self.rear]=a
             
    def delete(self):
        if self.front==-1:
            print("the queue is empty nothing to delete")
        elif self.front==self.rear :
            self.front = -1
            self.rear = -1
        else :
            self.front+=1
    
    def display(self):
        if self.front!=-1 :
         for i in range(self.front,self.rear+1):
            print(self.list[i])
        else :
            print("the queue is empty")
    
def main() :
    q = queue(5)
    q.insert(12)
    q.insert(2)
    q.insert(312)
    q.insert(162)
    q.insert(22)
    q.delete()
    q.delete()
    q.delete()
    q.delete()
    q.delete()
    q.display()
main()

File: data structures/queue/queue.java

public class queue {

    int[] arr = new int[6];
    int top = -1;
    int rear = -1;

    void insert(int a) {
        if (rear == arr.length - 1) {
            System.out.println("the queue is full");
        } else if (rear == -1) {
            top++;
            arr[top] = a;
            rear++;
        } else {

            rear++;
            arr[rear] = a;
        }
    }

    void delete() {
        if(front == rear)
      printf("\nQueue is Empty!!! Deletion is not possible!!!");
   else{
      printf("\nDeleted : %d", queue[front]);
      front++;
      if(front == rear)
	 front = rear = -1;
   }

    }

    void display() {
        if (top == -1 || top > rear) {
            System.out.println("the queue is empty");
            return;
        }
        for (int i = top; i <= rear; i++) {
            System.out.println(arr[i]);
        }
    }

    public static void main(String[] args) {
        queue o = new queue();
        o.insert(4);
        o.delete();
        o.display();
    }
}
